packages/tizen-remote/test/e2e/tizen-remote.e2e.spec.js

```diff
@@ -0,0 +1,343 @@
+/* eslint-disable promise/no-native */
+
+/**
+ * This test suite checks the interaction of the lib with a WebSocket server.
+ *
+ * If the following env variables are present, the test suite will use them:
+ *
+ * - `TEST_TIZEN_REMOTE_HOST`: The host of the WebSocket server.
+ * - `TEST_TIZEN_REMOTE_PORT`: The port of the WebSocket server.
+ *
+ * If these env variables are _not_ present, a mock server will be started on a random port.
+ *
+ * @module
+ */
+
+import {createSandbox} from 'sinon';
+import getPort from 'get-port';
+import {TestWSServer} from './server';
+import unexpected from 'unexpected';
+import {constants, Event, Keys, TizenRemote} from '../../lib/index';
+import d from 'debug';
+import unexpectedSinon from 'unexpected-sinon';
+import unexpectedEventEmitter from 'unexpected-eventemitter';
+import {Env} from '@humanwhocodes/env';
+
+const env = new Env();
+
+const HOST = /** @type {string} */(env.get('TEST_TIZEN_REMOTE_HOST', '127.0.0.1'));
+const PORT = env.get('TEST_TIZEN_REMOTE_PORT');
+
+const debug = d('tizen-remote:test:e2e:ws');
+
+const expect = unexpected.clone().use(unexpectedSinon).use(unexpectedEventEmitter);
+
+describe('websocket behavior', function () {
+  /** @type {TestWSServer} */
+  let server;
+
+  /** @type {number} */
+  let port;
+
+  /** @type {TizenRemote} */
+  let remote;
+
+  /** @type {sinon.SinonSandbox} */
+  let sandbox;
+
+  /** @type {import('../../lib').TizenRemoteOptions} */
+  let remoteOpts;
+
+  beforeEach(async function () {
+    sandbox = createSandbox();
+    if (PORT) {
+      port = Number(PORT);
+    } else {
+      port = await getPort();
+      server = new TestWSServer({
+        host: HOST,
+        port,
+        path: constants.API_PATH_V2,
+      });
+    }
+    remoteOpts = {
+      host: HOST,
+      port,
+      name: 'test',
+      token: 'hoboken',
+    };
+  });
+
+  afterEach(async function () {
+    sandbox.restore();
+    if (remote) {
+      try {
+        debug('[CLEANUP] Disconnecting from remote attached to %s:%d', HOST, port);
+        await remote.disconnect();
+      } catch {}
+    }
+    debug('[CLEANUP] Stopping server listening on %s:%d', HOST, port);
+    await server.stop();
+  });
+
+  it('should connect', async function () {
+    remote = new TizenRemote(remoteOpts);
+    await remote.connect();
+    expect(remote.isConnected, 'to be true');
+  });
+
+  describe('connection behavior', function () {
+    describe('token negotiation', function () {
+      describe('when the remote has no token', function () {
+        beforeEach(function () {
+          remoteOpts.token = undefined;
+          remote = new TizenRemote(remoteOpts);
+        });
+
+        it('should request a token from the server', async function () {
+          return await expect(
+            remote.connect(),
+            'to emit from',
+            remote,
+            Event.TOKEN,
+            `token-${remoteOpts.name}`
```

Comment:
this won't be true for a real token right?

Reply:
no, the real token is just like 8 integers.